src/components/CanvasWrapper.tsx

```diff
@@ -78,11 +78,16 @@ type CanvasWrapperProps = {
    * directly by calling `canv.setDataset()`.
    */
   dataset: Dataset | null;
+  datasetKey: string | null;
```

Comment:
can datasetKey always be gotten from dataset? why is it separate?

Reply:
It's the key of the dataset in the current collection. The dataset's metadata name is totally separate.